flocker/cli/functional/test_deploy_script.py

```diff
@@ -29,13 +44,91 @@ class FlockerDeployTests(TestCase):
+        self.persistence_service.startService()
+        self.cluster_state_service = ClusterStateService()
+        self.cluster_state_service.startService()
+        self.addCleanup(self.cluster_state_service.stopService)
+        self.addCleanup(self.persistence_service.stopService)
+        app = ConfigurationAPIUserV1(self.persistence_service,
+                                     self.cluster_state_service).app
+        self.port = reactor.listenTCP(0, Site(app.resource()),
+                                      interface="127.0.0.1")
+        self.addCleanup(self.port.stopListening)
+        self.port_number = self.port.getHost().port
 
     def test_version(self):
         """``flocker-deploy --version`` returns the current version."""
         result = check_output([b"flocker-deploy"] + [b"--version"])
         self.assertEqual(result, b"%s\n" % (__version__,))
 
+    def _send_configuration(self,
+                            application_config_yaml=COMPLEX_APPLICATION_YAML,
+                            deployment_config_yaml=COMPLEX_DEPLOYMENT_YAML):
+        """
+        Run ``flocker-deploy`` against the API server.
+
+        :param application_config: Application configuration dictionary.
+        :param deployment_config: Deployment configuration dictionary.
+
+        :return: ``Deferred`` that fires with a tuple (stdout, stderr,
+            exit code).
+        """
+        app_config = FilePath(self.mktemp())
+        app_config.setContent(safe_dump(application_config_yaml))
+        deployment_config = FilePath(self.mktemp())
+        deployment_config.setContent(safe_dump(deployment_config_yaml))
+        return getProcessOutputAndValue(
+            b"flocker-deploy", [
+                b"--nossh",
+                b"--port", unicode(self.port_number).encode("ascii"),
+                b"localhost", deployment_config.path, app_config.path],
+            env=environ)
+
+    def test_configures_cluster(self):
+        """
+        ``flocker-deploy`` sends the configuration to the API endpoint that
+        will replace the cluster configuration.
+        """
+        result = self._send_configuration()
+        apps = FlockerConfiguration(
+            deepcopy(COMPLEX_APPLICATION_YAML)).applications()
+        expected = model_from_configuration(
+            applications=apps,
+            deployment_configuration=deepcopy(COMPLEX_DEPLOYMENT_YAML))
+        result.addCallback(lambda _: self.assertEqual(
+            self.persistence_service.get(), expected))
+        return result
+
+    def test_output(self):
+        """
+        ``flocker-deploy`` prints a helpful message when it's done.
+        """
+        result = self._send_configuration()
+        result.addCallback(
+            self.assertEqual,
+            (b"The configuration has been updated. It may take a "
+             b"short while for changes to take effect, in "
+             b"particular if Docker images need to be pulled.\n",
```

Comment:
"... take effect; in particular, if Docker ..." would be close to correct, I think.  I'd be inclined to make this string a constant, too, and use the constant in both the implementation and the test.